### ISSUE TITLE
LOG_SCOPE_FUNCTION does not work properly on windows.

### ISSUE THREAD

**Issue by edmiester777** [emilk/loguru#43]
I use the example code and here is what outputs in windows:
![image](https://user-images.githubusercontent.com/1354397/31980332-c8a2cf22-b919-11e7-8307-b40ecde8e2d6.png)

And here is what outputs:
![image](https://user-images.githubusercontent.com/1354397/31980377-14deeace-b91a-11e7-9014-b302c7fd9f32.png)

It appears that the LOG_SCOPE_F macro immediately closes the scope while on windows.

Please note that I am using VS2017 to build.

**Comment by emilk** [comment]
Very weird. I don't have any access to Windows, so this is hard for me to debug. Anybody like to take a stab at it?

**Comment by edmiester777** [comment]
I'll take a crack at it.

**Comment by tomaszte** [comment]
Hi, got the same problem on windows. Done some tests:
```
//destructor called immediately
	LOG_SCOPE_F(INFO, "TEST.");

	//destructor called immediately
	loguru::LogScopeRAII error_context_RAII_TEST = ((loguru::Verbosity_INFO) > loguru::current_verbosity_cutoff()) ? loguru::LogScopeRAII() : loguru::LogScopeRAII(loguru::Verbosity_INFO, "win32_editor.cpp", 519, "TEST.");

	//destructor not called, indendation works fine
	loguru::LogScopeRAII error_context_RAII_TEST = loguru::LogScopeRAII(loguru::Verbosity_INFO, "win32_editor.cpp", 519, "TEST.");

```
So I guess windows treats it like that `(expression) ? {/*scope1*/ value1 } :  {/*scope2*/ value2 };`
Any idea for a fix?

**Comment by allisonvacanti** [comment]
@emilk I have a patch that fixes this on MSVC2015 x64. Implementing the move constructor and explicitly clearing the rvalue object's `_file` does the trick.

Just include `<utility>` for `std::move` and change the defaulted move constructor to:

```
    LogScopeRAII(LogScopeRAII&& other)
    {
      _verbosity = std::move(other._verbosity);
      _file = std::move(other._file);
      _line = std::move(other._line);
      _indent_stderr = std::move(other._indent_stderr);
      _start_time_ns = std::move(other._indent_stderr);
      memcpy(_name, other._name, LOGURU_SCOPE_TEXT_SIZE * sizeof(char));

      // Make sure the tmp object's destruction doesn't close the scope:
      other._file = nullptr;
    }
```

**Comment by emilk** [comment]
@allisonvacanti I appreciate your PR, but I absolutely want to avoid `#include`s in the Loguru header. Luckily there should be no need: all the fields can just be copied, removing the need for `std::move`. `_name` can be copied with a small for-loop, removing the need for `memcpy` (and thus `<cstring>`). I suggest you change your PR along these lines. Maybe we should also only define `LogScopeRAII(LogScopeRAII&& other)` for the compilers that need it (MSVC) to avoid potential overhead on other platforms?

**Comment by allisonvacanti** [comment]
Agreed -- I'd actually fixed this before seeing this PR, and @paralaxsd's solution is better, since we can avoid `std::move` using the initialization syntax (for some reason I wrote mine like an move assignment operator instead of a ctor).

I'll update the PR with your suggested changes and incorporate the earlier patch too.
